### PR TITLE
feat: Support selling entire balance

### PR DESCRIPTION
**Pull request by dekz**
# Description

For contracts where the balance is non-deterministic we want to sell what is available. 

Requires: https://github.com/0xProject/protocol/pull/37

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -93,3 +93,5 @@ TODOs.org
 test/docker-compose.yml
 
 TODO.txt
+
+scripts
```
Comment (kimpers) [inline .gitignore:97]: NIT: `~/.gitignore` is your friend personal ignores 😂

**Review by kimpers — APPROVED** [review]
LGTM 👍

**Comment by dekz** [comment]
deploy staging

**Comment by dekz** [comment]
deploy staging

**Pull request merged**